src/atlas_assistant/tools/dataset.py

```diff
@@ -0,0 +1,50 @@
+from langchain.tools import ToolRuntime, tool
+from langchain_core.messages import ToolMessage
+from langgraph.types import Command
+from pydantic import BaseModel
+
+from ..context import Context
+from ..dataset import Dataset, Metadata
+from ..settings import Settings
+from ..state import State
+
+
+class SearchResult(BaseModel):
+    """The result of an embeddings search"""
+
+    dataset: Dataset
+    """The dataset"""
+
+    score: float
+    """The search score"""
```

Comment:
Could this live next to the `Dataset` model and perhaps generalize that file to `models.py`?

Reply:
I don't think that makes sense, because the `Dataset` is a static bunch of metadata, whereas the score is generated by a similarity search against a specific query.

Reply:
Ya, I noticed the `sql` tool also has a model in there. Just trying to stay organized with my mental model of what's going where. I agree that it's fine where it is. 👍🏽